static/app/views/eventsV2/utils.tsx

```diff
@@ -542,6 +549,108 @@ export function shouldRenderPrebuilt(): boolean {
   const shouldRender = localStorage.getItem(RENDER_PREBUILT_KEY);
   return shouldRender === 'true' || shouldRender === null;
 }
+
 export function setRenderPrebuilt(value: boolean) {
   localStorage.setItem(RENDER_PREBUILT_KEY, value ? 'true' : 'false');
 }
+
+function eventViewToWidgetQuery({
+  eventView,
+  yAxis,
+  displayType,
+}: {
+  displayType: DisplayType;
+  eventView: EventView;
+  yAxis?: string[];
+}) {
+  const fields = eventView.fields.map(({field}) => field);
+  const {columns, aggregates} = getColumnsAndAggregates(fields);
+  const sort = eventView.sorts[0];
+  const widgetQuery: WidgetQuery = {
+    name: '',
+    aggregates: [
+      ...(displayType === DisplayType.TOP_N ? aggregates : []),
+      ...(typeof yAxis === 'string' ? [yAxis] : yAxis ?? ['count()']),
+    ],
+    columns: [...(displayType === DisplayType.TOP_N ? columns : [])],
+    fields: [
+      ...(displayType === DisplayType.TOP_N ? fields : []),
+      ...(typeof yAxis === 'string' ? [yAxis] : yAxis ?? ['count()']),
+    ],
+    conditions: eventView.query,
+    orderby: sort ? `${sort.kind === 'desc' ? '-' : ''}${sort.field}` : '',
+  };
+  return {fields, widgetQuery};
```

Comment:
It doesn't look like we use the `fields` return of this function, can we just drop it and have it return the widget query directly?